Fix `config.read_encrypted_secrets` deprecation warning quoting

viralpraxis opened the pull request:
[This commit ](https://github.com/rails/rails/commit/0c76f17f2dbf0d7ad90c890e6f334743cacce41f) deprecated `config.read_encrypted_secrets` but, surprisingly, deprecation used back-ticks instead of normal quoting (which leads to excuting string payload via shell).

### Motivation / Background

```
pry(main)> Rails::VERSION::STRING
"7.2.0.alpha"
pry(main)> Rails.application.config.read_encrypted_secrets
Errno::ENOENT: No such file or directory - config.read_encrypted_secrets
```

### Detail

If I'm not missing anything, normal quotes should be used.

### Additional information

After fix it works as expected:

```
[1] pry(main)> Rails.application.config.read_encrypted_secrets
ActiveSupport::DeprecationException: DEPRECATION WARNING: config.read_encrypted_secrets is deprecated and will be removed in Rails 7.3. (called from __pry__ at (pry):1)
```

<!-- Provide additional information such as benchmarks, references to other repositories, or alternative solutions. -->

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

akhilgkrishnan commented on railties/lib/rails/application/configuration.rb:
```diff
@@ -350,14 +350,13 @@ def enable_reloading=(value)
 
       def read_encrypted_secrets
-        Rails.deprecator.warn(`config.read_encrypted_secrets is deprecated and will be removed in Rails 7.3.`)
+        Rails.deprecator.warn("config.read_encrypted_secrets is deprecated and will be removed in Rails 7.3.")
```
```suggestion
        Rails.deprecator.warn("`config.read_encrypted_secrets` is deprecated and will be removed in Rails 7.3.")
```

We can highlight `config.read_encrypted_secrets` as a code snippet

flavorjones replied:
See my review comment, I think the community consensus is against backquotes in error/warning messages.

viralpraxis replied:
Replaced backquotes with single quotes

akhilgkrishnan commented on railties/lib/rails/application/configuration.rb:
```diff
@@ -350,14 +350,13 @@ def enable_reloading=(value)
 
       def read_encrypted_secrets=(value)
-        Rails.deprecator.warn(`config.read_encrypted_secrets is deprecated and will be removed in Rails 7.3.`)
+        Rails.deprecator.warn("config.read_encrypted_secrets is deprecated and will be removed in Rails 7.3.")
```
```suggestion
        Rails.deprecator.warn("`config.read_encrypted_secrets` is deprecated and will be removed in Rails 7.3.")
```

viralpraxis replied:
Thanks! Applied separately to keep it a single-commit PR

akhilgkrishnan commented on railties/lib/rails/application/configuration.rb:
```diff
@@ -350,14 +350,13 @@ def enable_reloading=(value)
 
       def read_encrypted_secrets=(value)
-        Rails.deprecator.warn(`config.read_encrypted_secrets is deprecated and will be removed in Rails 7.3.`)
+        Rails.deprecator.warn("`config.read_encrypted_secrets=` is deprecated and will be removed in Rails 7.3.")
```
```suggestion
        Rails.deprecator.warn("`config.read_encrypted_secrets` is deprecated and will be removed in Rails 7.3.")
```
typo?

viralpraxis replied:
I was not sure if `config.read_encrypted_secrets` (without `=`) was intentional is this case, since actually `config.read_encrypted_secrets=` is called. Do you think warning with `config.read_encrypted_secrets` in both cases is more appropriate?

flavorjones replied:
I think keeping the `=` in this case is reasonable.

flavorjones reviewed (changes requested):
I think the backquotes should be removed entirely or replaced with single-quotes.

There was a recent ruby-core discussion about error messages and the consensus was that backticks in messages often confuse parsers (e.g., markdown) when pasted into forums and other sites: https://bugs.ruby-lang.org/issues/16495#note-32

viralpraxis committed:
Fix config.read_encrypted_secrets deprecation warning quoting

Commit 0c76f17 deprecated `Rails::Application::Configuration#read_encrypted_secrets`
but, suprisingly, deprecation used backticks instead of normal quoting.

flavorjones commented:
Note that I've opened a PR with a related rubocop style cop enabled: #51741

akhilgkrishnan commented:
> https://bugs.ruby-lang.org/issues/16495#note-32

Thanks @flavorjones for sharing this
